### DISCUSSION TITLE
Understanding memory usage

### DISCUSSION THREAD

**Discussion by siddhsql** [ggml-org/llama.cpp#1876] [Q&A] [answered]
I am running this program on a Mac. When I load a model, I see following in the output:

<img width="946" alt="image" src="https://github.com/ggerganov/llama.cpp/assets/127623723/d3dbbf06-de4c-4079-bf76-a17110deef4a">

this shows expected memory usage as 11359 MB. But when I inspect in activity monitor I see:

<img width="819" alt="image" src="https://github.com/ggerganov/llama.cpp/assets/127623723/d44779d2-224e-467d-80a0-db26b2ba7afc">

so memory consumed is 2.4 GB as per activity monitor. can someone explain the discrepancy to me? thanks.

**Comment by africalimedrop** [comment, marked as answer]
Suspect this will help answer; bolded relevant part:

-   `--no-mmap`: Do not memory-map the model. **By default, models are mapped into memory, which allows the system to load only the necessary parts of the model as needed.** However, if the model is larger than your total amount of RAM or if your system is low on available memory, using mmap might increase the risk of pageouts, negatively impacting performance. Disabling mmap results in slower load times but may reduce pageouts if you're not using `--mlock`. Note that if the model is larger than the total amount of RAM, turning off mmap would prevent the model from loading at all.

**Comment by siddhsql** [comment]
thanks. makes sense.
